modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/service/impl/KBArticleLocalServiceImpl.java

```diff
@@ -2209,24 +2351,35 @@ private void _validateUrlTitle(
 	@Reference
 	private KBFolderPersistence _kbFolderPersistence;
 
+	@Reference
+	private OrganizationLocalService _organizationLocalService;
+
 	@Reference
 	private Portal _portal;
 
 	@Reference
 	private PortletFileRepository _portletFileRepository;
 
+	private Date _previousCheckDate;
+
```

Comment:
This is a no no. You should never ever store state between invocations like this. Think of what happens if the component is deactivated/activated for whatever reason.
Also, there's a race condition (what would happen if I set the check interval to 1 minute?).

Reply:
Also, I don't see the point of storing this date? What do you need it for?

Reply:
I followed the pattern of other places (journal, FileEntry)

Reply:
not needed "by now"